src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -463,85 +467,127 @@ private Type EmitIListCoercion<T>(AssociativeNode arg)
-                    foreach (var asm in assemblies)
-                    {
-                        var type = asm.GetType(className);
-                        if (type == null) continue;
-
-                        // There should be a way to get the exact method after matching parameter types for a node
-                        // using its function descriptor. AST isn't sufficient for parameter type info.
-                        // Fist check for static methods
-                        mbs = type.GetMethods(BindingFlags.Public | BindingFlags.Static).Where(
-                            m => m.Name == methodName && m.GetParameters().Length == args.Count).ToList();
-
-                        // Check for instance methods
-                        if (mbs == null || !mbs.Any())
-                        {
-                            mbs = type.GetMethods(BindingFlags.Public | BindingFlags.Instance).Where(
-                                m => m.Name == methodName && m.GetParameters().Length + 1 == args.Count).ToList();
-                        }
-
-                        // Check for property getters
-                        if (mbs == null || !mbs.Any())
-                        {
-                            var prop = type
-                                .GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance).Where(
-                                    p => p.Name == methodName).FirstOrDefault();
-
-                            if (prop != null)
-                            {
-                                mbs = prop.GetAccessors().ToList();
-                            }
-                        }
-
-                        // Check for constructorinfo objects
-                        if(mbs == null || !mbs.Any())
-                        {
-                            mbs = type.GetConstructors().Where(
-                                m => m.DeclaringType.Name == methodName && m.GetParameters().Length == args.Count).ToList();
-                        }
-
-                        if (mbs != null && mbs.Any())
-                        {
-                            methodCache.Add(key, mbs);
-                            break;
-                        }
+                    methodName = ProtoCore.DSASM.Constants.kGetterPrefix + methodName;
+                    fg = runtimeCore.GetFuncGroup(methodName, className);
+                }
 
-                        //if (method != null)
-                        //{
-                        //    argTypes = method.GetParameters().Select(p => p.ParameterType).ToList();
-                        //    return method.ReturnType;
-                        //}
-                    }
+                // Try to find a setter
+                if (fg == null)
+                {
+                    methodName = ProtoCore.DSASM.Constants.kSetterPrefix + methodName;
+                    fg = runtimeCore.GetFuncGroup(methodName, className);
```

Comment:
No harm with this change but I thought we didn't support setters. At least I haven't seen any being used.

Reply:
For some reason I thought they were handled before..(but that may just have been the getters)
I can remove it if that is the case